Tavisca.Bootcamp.LanguageBasics.Exercise3/Program.cs

```diff
@@ -28,20 +29,164 @@ static void Main(string[] args)
-            Console.WriteLine(result);
+            Console.WriteLine(result);
+        }
+
+        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
+        {
+            var calorie = new int[protein.Length];
+            var dietList = new List<int>();
+            for(int i=0;i<protein.Length;i++)
+            {
+                calorie[i] = (protein[i] + carbs[i])*4 + fat[i]*9;
+            }
+            var index = new List<int>();
+            foreach (var item in dietPlans)
+            {
+                index.Clear();
+               if(string.IsNullOrWhiteSpace(item))
+                {
+                    index.Clear();
+                    index.Add(0);
+                }
+               else
+                {
+                    foreach (var ch in item)//CcPpFfTt
+                    {
+                        if (index.Count == 1)
+                            break;
+                        if (ch == 'C')
+                        {
+                            FindMaxIndex(carbs, index);
+                        }
+                        else if (ch == 'c')
+                        {
+                            FindMinIndex(carbs, index);
+                        }
+                        else if (ch == 'P')
+                        {
+                            FindMaxIndex(protein, index);
+                        }
+                        else if (ch == 'p')
+                        {
+                            FindMinIndex(protein, index);
+                        }
+                        else if (ch == 'F')
+                        {
+                            FindMaxIndex(fat, index);
+                        }
+                        else if (ch == 'f')
+                        {
+                            FindMinIndex(fat, index);
+                        }
+                        else if (ch == 'T')
+                        {
+                            FindMaxIndex(calorie, index);
+                        }
+                        else if (ch == 't')
+                        {
+                            FindMinIndex(calorie, index);
+                        }
+                    }
+                }
+                dietList.Add(index[0]);
+            }
+            int[] diet = new int[dietPlans.Length];
+            dietList.CopyTo(diet);
+            dietList.Clear();
+            return diet;
         }
 
-        public static int[] SelectMeals(int[] protein, int[] carbs, int[] fat, string[] dietPlans)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        private static void FindMinIndex(int[] arr, List<int> index)
+        {
+            var list = new List<int>();
+            var minValue = int.MaxValue;
+            if(index.Count==0)
+            {
+                list.Clear();
+                for(int i=0;i<arr.Length;i++)

```

Comment:
The same logic is repeated. Please try to create a new common method to avoid code repetition.